### PR TITLE
chore: fix ci tests in winter24 (backport)

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Safari 16.6 introduced a breakage to one of our tests. Rather than implement the fix (#3678) in winter24, this PR attempts to do the minimum possible to fix CI tests without changing the actual code.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

**Comment by nolanlawson** [comment]
SauceLabs does not support minor browser versions, so we need to use Safari 15 instead of Safari 16.4.

Also, our benchmark smoke tests were broken due to a new version of Chromedriver not supporting Node 16.

**Comment by nolanlawson** [comment]
I can't get the WebDriverIO or Benchmark-Smoke tests to work. Too many breaking changes due to Chrome, Node, WebDriverIO, etc.

File: .circleci/config.yml [hunk 1]
```diff
@@ -319,25 +321,28 @@ jobs:
...
+    executor: node
+    environment:
+      SAUCE_TUNNEL_ID: lwc_<< pipeline.id >>_test_integration_dev
     steps:
-      - browser-tools/install-browser-tools # install local chrome rather than using saucelabs
       - load_workspace
+      - start_sauce_connect
       - retry_command:
-          command_name: Run integration test - local Chrome - dev mode
-          command: yarn local:dev
```
Comment (nolanlawson) [inline .circleci/config.yml:340]: Running integration tests in SauceLabs instead of local Chrome – we did this in `master` as well

**Comment by nolanlawson** [comment]
/nucleus ignore --reason 'some downstreams are borked, and nucleus has never worked for patch branches anyway'

**Review by jmsjtu — APPROVED** [review]

**Pull request merged**